test/Microsoft.TemplateEngine.Authoring.TemplateVerifier.IntegrationTests/ExampleTemplateTest.cs

```diff
@@ -33,10 +38,47 @@ public async void VerificationEngineSampleDogfoodTest()
                 SnapshotsDirectory = "Snapshots",
                 OutputDirectory = workingDir,
                 VerifyCommandOutput = true,
-                DoNotPrependTemplateNameToScenarioName = true,
+                DoNotPrependCallerMethodNameToScenarioName = true,
                 DoNotAppendTemplateArgsToScenarioName = true,
+                ScenarioName = "SampleDogfoodTest",
+                // This is here just for testing and documentation purposes - showing functionality of differing snapshots for arch
+                UniqueFor = UniqueForOption.Architecture,
+            }
+                .WithCustomScrubbers(
+                    ScrubbersDefinition.Empty
+                        .AddScrubber(sb => sb.Replace("B is enabled", "*******"))
+                        .AddScrubber((path, content) =>
+                        {
+                            if (path.Replace(Path.DirectorySeparatorChar, '/') == "std-streams/stdout.txt")
+                            {
+                                content.Replace("SampleTestTemplate", "%TEMPLATE%");
+                            }
+                        }));
+
+            VerificationEngine engine = new VerificationEngine(_log);
+            await engine.Execute(options);
+        }
+
+        [Fact]
+        public async void VerificationEngineSampleDogfoodTest_ExecThroughApi()
+        {
+            string workingDir = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName().Replace(".", string.Empty));
+            string templateShortName = "TestAssets.SampleTestTemplate";
+
+            //get the template location
+            string templateLocation = Path.Combine(TestTemplatesLocation, "TestTemplate");
+
+            TemplateVerifierOptions options = new TemplateVerifierOptions(templateName: templateShortName)
+            {
+                TemplatePath = templateLocation,
+                SnapshotsDirectory = "Snapshots",
+                OutputDirectory = workingDir,
+                VerifyCommandOutput = true,
+                DoNotPrependCallerMethodNameToScenarioName = true,
+                ScenarioName = "SampleDogfoodTest",
                 UniqueFor = UniqueForOption.Architecture,
             }
+                .WithInstantiationThroughTemplateCreatorApi(new Dictionary<string, string?>() { { "paramB", "true" } })
```

Comment:
@YuliiaKovalova this can be inspiration how the imported sample templates might be tested (only `templateName`, `TemplatePath` and `WithInstantiationThroughTemplateCreatorApi` will be needed - rest is here to unit test the actual testing functionality)